src/charts/pie.coffee

```diff
@@ -292,8 +292,17 @@ Ember.Charts.PieComponent = Ember.Charts.ChartComponent.extend(
   labelAttrs: Ember.computed ->
     arc = @get 'arc'
     labelRadius = @get 'labelRadius'
-    lastXPos = 0
-    lastYPos = 0
+    # these are the label regions that are already filled
+    usedLabelPositions =
+      left: []
+      right: []
+    # assumes height of all the labels are the same
+    labelOverlap = (side, ypos, height) ->
+      positions = usedLabelAreas[side]
+      for pos in positions
+        if Math.abs(ypos - pos) - height
+          return yes
```

Comment:
Can be one line (`return yes if Math.abs(ypos - pos) - height`), though not sure that matters too much